sleap/gui/app.py

```diff
@@ -1364,15 +1364,15 @@ def updateStatusMessage(self, message: Optional[str] = None):
                     )
                     message += " in video"
 
-            lf = self.state["labeled_frame"]
-            n_instances = 0 if lf is None else len(lf)
+            visible_instances = self.state["labeled_frame"].instances_to_show
```

Comment:
Is `self.state["labeled_frame"]` always defined? I thought it was initialized to `None`, which is why we were doing that check above.

If it's `None`, then doing `self.state["labeled_frame"].instances_to_show` will give you an attribute error saying that `NoneType` objects don't have an `instances_to_show` attribute.

Reply:
One more thing: this doesn't require a change necessarily, but I'd discourage using the `LabeledFrame.instances_to_show` property.

The logic is that it returns all user instances plus all predicted instances that are "unused" ([here](https://github.com/murthylab/sleap/blob/a2738ec10ee5dd2c16df678edfddc1d0eb8aadc8/sleap/instance.py#L1544)).

This is meant to address the case where we have predicted instances that we double-clicked to create a user instance from. In that case, we'd want to not display those since the user instances will be overlapping with them.

The issue is that our logic for finding the "unused" predicted instances is a little weird. It's defined in the `LabeledFrame.unused_predictions` property, in which there's a note about the approach we use ([here](https://github.com/murthylab/sleap/blob/a2738ec10ee5dd2c16df678edfddc1d0eb8aadc8/sleap/instance.py#L1520)).

The weirdness comes from the fact that we use the `PredictedInstance.track` to both determine whether it's "unused" and also to sort the order they're returned in, even though the presence of a track has no bearing on whether the user has double-clicked it or not. I think this was a kludgy fix to address #518, but some of the logic might pre-date even that.

Given this mess, we may want to change how all of this works in the future, so it might be a good idea to avoid using if possible (or at least dropping TODO comment mentioning that it might need to be revisited in the future).

Reply:
Updated this to one again check if `self.state['labeled_frame'] is None`, but still use `LabeledFrame.instances_to_show`. However, I added a note to revisit this line if either the code for `LabeledFrame.instances_to_show` or `LabeledFrame.unused_predictions` changes.